Frontend/student-hub/src/App.js

```diff
@@ -25,30 +29,34 @@ function App() {
   }, []);
 
   return (
-
+    <ThemeProvider>
     <Router>
       <Navbar isLoggedIn={isLoggedIn} setIsLoggedIn={setIsLoggedIn} />
       <Container maxWidth="sm">
       <Grid container spacing={2} alignItems="center" justify="center" style={{ minHeight: '100vh' }}>
             <Grid item xs={12}>
+            <Snackbar open={!!error} autoHideDuration={6000} onClose={() => setError('')}>
+              <Alert severity="error">{error}</Alert>
+            </Snackbar>
       <Routes>
       <Route path="/" element={<Home />} />
-      <Route path="/login" element={<Login setIsLoggedIn={setIsLoggedIn} />} />
-      <Route path="/signup" element={<Signup setIsLoggedIn={setIsLoggedIn} />} />
-      <Route path="/dashboard" element={<Dashboard />} />
-      <Route path="/profile" element={<Profile />} />
-      <Route path="/courses" element={<CoursePage />} />
-      <Route path="/mentorship" element={<MentorshipPage />} />
+      <Route path="/login" element={<Login setIsLoggedIn={setIsLoggedIn}  setError={setError} error={error} />} />
```

Comment:
ah this is interesting, so this pattern of passing around a singular prop from a parent component and letting it get accessed in multiple children or grandchildren components is generally through the usage of Context. You can read more about that here: https://react.dev/learn/passing-data-deeply-with-context

This actually something you're already doing with the Theme! so shouldn't be too unfamiliar

Reply:
Thanks for drawing my attending to this, I have addressed this feedback.